Source/WebCore/platform/network/soup/ResourceRequestSoup.cpp

```diff
@@ -140,6 +140,18 @@ void ResourceRequest::updateSoupMessageBody(SoupMessage* soupMessage, BlobRegist
 
 }
 
+GRefPtr<GBytes> ResourceRequest::createBodyBytes() const
+{
+    auto formData = httpBody();
+    if (!formData || formData->isEmpty())
+        return nullptr;
+
+    auto bytes = formData->resolveBlobReferences()->flatten();
```

Comment:
I don't think this is a good idea, flatten ignores blobs and files. I'm not sure if it would be possible to have a request body with blobs and files for custom uri schemes, though.

Reply:
hmm, but if the form data can contain files... then the stream approach was better... I'm sorry I didn't realized

Reply:
I hope it can be a stream type if that can happen. In my previous implementation, I used `webkitFormDataInputStreamNew`, which will get the type `WebKitFormDataInputStream`. This should behave like a stream, right? If you can agree, I can fall back to the previous commit that returns input stream.

Reply:
Yes, sorry, I think the safest approach would be to return a stream like in your first patch